browser/farbling/brave_screen_farbling_browsertest.cc

```diff
@@ -0,0 +1,270 @@
+
+    host_resolver()->AddRule("*", "127.0.0.1");
+    content::SetupCrossSiteRedirector(embedded_test_server());
+
+    brave::RegisterPathProvider();
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    embedded_test_server()->ServeFilesFromDirectory(test_data_dir);
+
+    ASSERT_TRUE(embedded_test_server()->Start());
+
+    top_level_page_url_ = embedded_test_server()->GetURL("a.com", "/");
+    farbling_url_ = embedded_test_server()->GetURL("a.com", "/simple.html");
+  }
+
+  void TearDown() override {
+    browser_content_client_.reset();
+    content_client_.reset();
+  }
+
+  HostContentSettingsMap* content_settings() {
+    return HostContentSettingsMapFactory::GetForProfile(browser()->profile());
+  }
+
+  void AllowFingerprinting() {
+    brave_shields::SetFingerprintingControlType(
+        content_settings(), ControlType::ALLOW, top_level_page_url_);
+  }
+
+  void SetFingerprintingDefault() {
+    brave_shields::SetFingerprintingControlType(
+        content_settings(), ControlType::DEFAULT, top_level_page_url_);
+  }
+
+  content::WebContents* contents() const {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  bool NavigateToURLUntilLoadStop(const GURL& url) {
+    EXPECT_TRUE(ui_test_utils::NavigateToURL(browser(), url));
+    return WaitForLoadStop(contents());
+  }
+
+  Browser* OpenPopup(const std::string& script) const {
+    content::ExecuteScriptAsync(contents(), script);
+    Browser* popup = ui_test_utils::WaitForBrowserToOpen();
+    EXPECT_NE(popup, browser());
+    auto* popup_contents = popup->tab_strip_model()->GetActiveWebContents();
+    EXPECT_TRUE(WaitForRenderFrameReady(popup_contents->GetMainFrame()));
+    return popup;
+  }
+
+  const GURL& farbling_url() { return farbling_url_; }
+
+ protected:
+  base::test::ScopedFeatureList feature_list_;
+
+ private:
+  GURL top_level_page_url_;
+  GURL farbling_url_;
+  std::unique_ptr<ChromeContentClient> content_client_;
+  std::unique_ptr<BraveContentBrowserClient> browser_content_client_;
+};
+
+class BraveScreenFarblingBrowserTest_DisableFlag
+    : public BraveScreenFarblingBrowserTest {
+ public:
+  BraveScreenFarblingBrowserTest_DisableFlag() {
+    feature_list_.InitAndDisableFeature(
+        blink::features::kBraveBlockScreenFingerprinting);
+  }
+};
+
+const gfx::Rect testWindowBounds[] = {
+    gfx::Rect(50, 50, 100, 100), gfx::Rect(50, 50, 100, 0),
+    gfx::Rect(200, 100, 0, 100), gfx::Rect(-100, -200, 20000, 10000),
+    gfx::Rect(0, 0, 0, 0)};
+
+const char* testScreenSizeScripts[] = {
+    "window.outerWidth - window.innerWidth",
+    "window.outerHeight - window.innerHeight",
+    "window.screen.availWidth - window.innerWidth",
+    "window.screen.availHeight - window.innerHeight",
+    "window.screen.width - window.innerWidth",
+    "window.screen.height - window.innerHeight",
+};
+
+#define FARBLE_SCREEN_SIZE(disable_flag)                                    \
```

Comment:
why are you using defines instead of methods for this stuff?

Reply:
Good point -- fixed.